core/src/avm2/globals/array.rs

```diff
@@ -275,20 +285,28 @@ impl<'gc> ArrayIter<'gc> {
 
         Ok(Self {
             array_object,
-            index: 0,
-            length,
+            index: if start_index < length {
+                start_index
+            } else {
+                length
+            },
+            rev_index: if end_index < length {
```

Comment:
This is equivalent to `(end_index + 1).min(length)`.

Reply:
This actually needed to be a saturating add, as one of the default parameters for end parameters on vector functions is `u32::MAX`.

I have to wonder if this is still subtly wrong if you have a vector with 4 billion elements, but at that point you've probably filled memory anyway.